### PR TITLE
Home w

### PR DESCRIPTION
**Pull request by IParov**

### CODE REVIEW

File: src/main/java/core/basesyntax/Colors.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public enum Colors {
+    red, yellow, green, white, blue, violet, orange
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Colors.java:4]: https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names

File: src/main/java/core/basesyntax/Colors.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public enum Colors {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Colors.java:3]:
```suggestion
public enum Color {
```

File: src/main/java/core/basesyntax/Figure.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+public abstract class Figure implements AreaCalculator {
+
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Figure.java:4]: empty line

File: src/main/java/core/basesyntax/Figure.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    @Override
+    public String toString() {
+        return "Figure: ";
+    }
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Figure.java:22]: why do we need this?

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int BOUND = 5;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/FigureSupplier.java:6]: What is 5, can you explain? It should be clear from its name.

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    private static final int PARAM_BOUND = 25;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/FigureSupplier.java:7]: max length?

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+                        random.nextInt(PARAM_BOUND));
+            default:
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        random.nextInt(PARAM_BOUND),
+                        random.nextInt(PARAM_BOUND));
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle("white", 10);
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/FigureSupplier.java:34]: "white" and 10 should be constants

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+public class FigureSupplier {
+    private static final int BOUND = 5;
+    private static final int PARAM_BOUND = 25;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(BOUND)) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(), random.nextInt(PARAM_BOUND));
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/FigureSupplier.java:14]: better create methods like `getRandomCircle()` for each figure type

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Figure.java (hunk 4 above)
Comment (IParov) [reply src/main/java/core/basesyntax/Figure.java:22]: Hello. According to the task, figures should be printed out with the word "Figure " before. Please check the task description.   Or I can add this word to each class toString method.

File: src/main/java/core/basesyntax/FigureSupplier.java (hunk 6 above)
Comment (IParov) [reply src/main/java/core/basesyntax/FigureSupplier.java:7]: I will change the names of these two variables to be more readable if this is what you mean.

File: src/main/java/core/basesyntax/Circle.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private static final int DEFAULT_RADIUS = 10;
+    private static final String DEFAULT_COLOR = Color.WHITE.name().toLowerCase();
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Circle.java:5]: those should be constants in FigureSupplier

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+                random.nextInt(FIGURE_PARAMETERS_BOUND),
+                random.nextInt(FIGURE_PARAMETERS_BOUND),
+                random.nextInt(FIGURE_PARAMETERS_BOUND));
+    }
+
+    public Figure getRandomRectangle() {
+        return new Rectangle(colorSupplier.getRandomColor(),
+                random.nextInt(FIGURE_PARAMETERS_BOUND),
+                random.nextInt(FIGURE_PARAMETERS_BOUND));
+    }
```
Comment (aaananas) [inline src/main/java/core/basesyntax/FigureSupplier.java:36]: those should be private methods

File: src/main/java/core/basesyntax/Circle.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    public double getArea() {
+        return Math.PI * Math.pow(radius, 2);
+    }
+
+    @Override
+    public String toString() {
+        return "Figure: Circle, " + "area: " + getArea() + " sq.units, "
+                + "radius: " + radius + " units, " + "color: " + getColor();
+    }
+}
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Circle.java:28]: image

**Review by aaananas — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Circle.java [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+    @Override
+    public String toString() {
+        return "Figure: Circle, " + "area: " + getArea() + " sq.units, "
+                + "radius: " + radius + " units, " + "color: " + getColor();
+    }
+
+    @Override
+    public String draw() {
+        return toString();
+    }
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Circle.java:25]: Do not override toString and change draw()

image

File: src/main/java/core/basesyntax/Figure.java (hunk 4 above)
Comment (Rommelua) [reply src/main/java/core/basesyntax/Figure.java]: image

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+
+public class FigureSupplier {
+    private static final int DEFAULT_RADIUS = 10;
+    private static final String DEFAULT_COLOR = Color.WHITE.name().toLowerCase();
+    private static final int RANDOM_FIGURE_COUNTER_BOUND = 5;
+    private static final int FIGURE_PARAMETERS_BOUND = 25;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    private Figure getRandomCircle() {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/FigureSupplier.java:13]: Private ones (methods, classes) should be after public ones.

File: src/main/java/core/basesyntax/Drawable.java [hunk 14]
```diff
@@ -0,0 +1,6 @@
+package core.basesyntax;
+
+public interface Drawable {
+
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Drawable.java:4]: empty line

File: src/main/java/core/basesyntax/Drawable.java [hunk 15]
```diff
@@ -0,0 +1,6 @@
...
+    String draw();
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/Drawable.java:5]:
```suggestion
    void draw();
```

**Review by Rommelua — CHANGES_REQUESTED** [review]

**Review by Rommelua — APPROVED** [review]

**Pull request closed without merging**